DDG4/CMakeLists.txt

```diff
@@ -8,68 +8,131 @@
 # For the list of contributors see $DD4hepINSTALL/doc/CREDITS.
 #
 #==========================================================================
-dd4hep_package(    DDG4
-  USES             DDCore
-                   [GEANT4 REQUIRED COMPONENTS gdml ui_all vis_all]
-                   [ROOT   REQUIRED COMPONENTS Geom GenVector RIO]
-                   BOOST
-  OPTIONAL         ${CLHEP} [XercesC QUIET]
-  INCLUDE_DIRS     include
-  INSTALL_INCLUDES include/DDG4 )
-dd4hep_print ( "|++> CLHEP_LIBRARIES ${CLHEP_LIBRARIES} ${CLHEP_LIBRARY}" )
+SET_PROPERTY(DIRECTORY . PROPERTY PACKAGE_NAME DDG4)
+
+# configure Geant4
+IF(NOT DD4HEP_USE_GEANT4)
+  dd4hep_print("Not Using geant4, not building DDG4")
+  RETURN()
+ENDIF()
+
 #---Add Library---------------------------------------------------------------------
-dd4hep_add_package_library(DDG4 SOURCES src/*.cpp )
+file(GLOB DDG4_SOURCES ${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp)
+add_library(DDG4 SHARED ${DDG4_SOURCES})
+
+target_link_libraries(DDG4
+  PUBLIC
+  DDCore
+  Geant4::Interface
+  )
+
+#Ensure our own includes come before those of the system
+target_include_directories(DDG4 BEFORE
+  PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/>
+  $<INSTALL_INTERFACE:include>
+  $<TARGET_PROPERTY:DDCore,INTERFACE_INCLUDE_DIRECTORIES>  # FIXME: Check if this works when using imported target
+)
+
 #---------------------------  Legacy libraries (for Frank) -------------------------
-dd4hep_add_plugin(DDG4Legacy    SOURCES legacy/*.cpp)
+dd4hep_add_plugin(DDG4Legacy  SOURCES legacy/*.cpp USES DDG4)
 #-----------------------------------------------------------------------------------
 dd4hep_add_dictionary( G__DDG4
-  SOURCES ${DDCore_INCLUDE_DIRS}/include/ROOT/Warnings.h python/DDG4Dict.C
-  LINKDEF ${DDCore_INCLUDE_DIRS}/include/ROOT/LinkDef.h
+  SOURCES python/DDG4Dict.C
+  USES    DDCore DDParsers DDG4 Geant4::Interface
   )
 #---------------------------  Plugin library for the simulation framework  ---------
 dd4hep_add_plugin(DDG4Plugins
-  GENERATED   G__DDG4.cxx
-  SOURCES     plugins/*.cpp
+  SOURCES   plugins/*.cpp
+  GENERATED G__DDG4.cxx
+  USES      ROOT::Core ${CLHEP} DDG4 ${XML_LIBRARIES} DDParsers
   )
 #---------------------------  Plugin library for the simulation framework  ---------
-dd4hep_add_dictionary(G__DDG4Python
-  SOURCES ${DDCore_INCLUDE_DIRS}/include/ROOT/Warnings.h src/python/DDG4Python.C
-  LINKDEF ${DDCore_INCLUDE_DIRS}/include/ROOT/LinkDef.h
-  OPTIONAL    [PYTHON REQUIRED]
-  )
+
+
+find_package(PythonInterp 2)
+find_package(PythonLibs 2)
```

Comment:
Since we agreed to 3.12 `find_package (Python COMPONENTS Interpreter Development)`

Reply:
Changed to
```
find_package(Python2 COMPONENTS Development)
```
Interpreter is not used, as far as I can tell, and this was only needed because the find_package for pythonlibs was failing otherwise.
Once python3 support is possible we can drop the 2, or make the user explicitly choose which python version to use.